code/modules/mob/living/say.dm

```diff
@@ -62,6 +62,24 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 /proc/prefix_to_channel(prefix)
 	return GLOB.department_radio_keys[prefix]
 
+/proc/filter_message(user, message)
+	var/list/banned_words = CONFIG_GET(str_list/word_filter)
+	var/message_lowercase = lowertext(message)
+
+	for(var/word in banned_words)
+		if(findtext(message_lowercase, regex(lowertext(word))))
+			to_chat(user,
+				html = "\n<font color='red' size='4'><b>-- Word Filter Message --</b></font>",
+				)
+			to_chat(user,
+				type = MESSAGE_TYPE_ADMINPM,
+				html = "\n<font color='red' size='4'><b>Your message has been automatically filtered due to its contents. Trying to circumvent this filter will get you banned.</b></font>",
+				)
+			SEND_SOUND(user, sound('sound/effects/adminhelp_new.ogg'))
+			return FALSE
```

Comment:
Have it send a message to admins maybe?

Reply:
The entire point is this is a preliminary measure to admin intervention